src/connectors/ledger.js

```diff
@@ -30,42 +32,40 @@ export class LedgerConnector extends AbstractConnector {
     accountFetchingConfigs,
     baseDerivationPath
   }) {
-    super({
-      supportedChainIds: [chainId]
-    })
+    super({ supportedChainIds: [chainId] })
+
     this.chainId = chainId
     this.url = url
     this.pollingInterval = pollingInterval
     this.requestTimeoutMs = requestTimeoutMs
     this.accountFetchingConfigs = accountFetchingConfigs
     this.baseDerivationPath = baseDerivationPath
-
-    this.ledgerSubprovider = null
   }
 
-  /**
-   * @returns {Promise<Object>}
-   */
-  async activate(): Promise {
+  async activate(): Promise<ConnectorUpdate> {
     if (!this.provider) {
-      const engine = new ProviderEngine();
-
-      const getTransport = () => TransportU2F.create();
-      const ledger = createLedgerSubprovider(getTransport, {
-        accountsLength: 1,
-        networkId: await this.getChainId()
-      })
-
-      this.ledgerSubprovider = ledger
-      
-      engine.addProvider(ledger)
-
-      engine.addProvider(new WebsocketSubprovider({ 
-        rpcUrl: this.url 
-      }))
-
-      engine.start()
+      let ledgerEthereumClientFactoryAsync = async () => {
+        const ledgerConnection = await TransportU2F.create()
+        // Ledger will automatically timeout the U2F "sign" request after `exchangeTimeout` ms.
+        // This will result in a cryptic error:
+        //   `{name: "TransportError", message: "Failed to sign with Ledger device: U2F DEVICE_INELIGIBLE", ...}`
+        // Setting the exchange timeout fixes that, although I haven't seen it documented anywhere else in the Ledger docs.
+        ledgerConnection.setExchangeTimeout(100000)
+        const ledgerEthClient = new AppEth(ledgerConnection)
+        return ledgerEthClient
+      }
 
+      const engine = new Web3ProviderEngine({ pollingInterval: this.pollingInterval })
+      engine.addProvider(
+        new LedgerSubprovider({
+          networkId: this.chainId,
+          ledgerEthereumClientFactoryAsync,
```

Comment:
What a property name.

Reply:
I mean it's no AbstractSingletonProxyFactoryBean... 😄